stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/DefaultChatEventHandler.kt

```diff
@@ -6,82 +6,79 @@ import io.getstream.chat.android.client.api.models.QueryChannelsRequest
+            .map { channels -> channels.any { it.cid == cid } }
             .let { it.isSuccess && it.data() }
     }
 
     override fun onNotificationAddedToChannelEvent(
         event: NotificationAddedToChannelEvent,
         filter: FilterObject,
     ): EventHandlingResult =
-        handleCidEventByRequest(event, filter)
+        handleCidEventByRequest(event.cid, filter)
 
     override fun onChannelUpdatedByUserEvent(
         event: ChannelUpdatedByUserEvent,
         filter: FilterObject,
-    ): EventHandlingResult =
-        handleCidEventByRequestIfNeeded(event, filter)
+    ): EventHandlingResult = EventHandlingResult.SKIP
 
     override fun onChannelUpdatedEvent(event: ChannelUpdatedEvent, filter: FilterObject): EventHandlingResult =
-        handleCidEventByRequestIfNeeded(event, filter)
+        EventHandlingResult.SKIP
 
     /**
-     * Handles [NotificationMessageNewEvent]. If the current channel list contains the channel from this event then it
-     * returns [EventHandlingResult.SKIP] otherwise it makes a request to API to define outcome of handling.
+     * Handles [NotificationMessageNewEvent]. It makes a request to API to define outcome of handling.
      *
      * @param event Instance of [NotificationMessageNewEvent] that is being handled.
      * @param filter [FilterObject] which is used to define an outcome.
      */
     override fun onNotificationMessageNewEvent(
         event: NotificationMessageNewEvent,
         filter: FilterObject,
-    ): EventHandlingResult {
-        return if (channels.value.any { it.cid == event.cid }) {
-            EventHandlingResult.SKIP
-        } else {
-            handleCidEventByRequest(event, filter)
-        }
-    }
+    ): EventHandlingResult = handleCidEventByRequest(event.cid, filter)
 
-    private fun handleCidEventByRequestIfNeeded(event: HasChannel, filter: FilterObject): EventHandlingResult {
-        return if (checkFilterOnChannelUpdatedEvent) {
-            handleCidEventByRequest(event, filter)
-        } else {
-            EventHandlingResult.SKIP
-        }
-    }
+    /**
+     * Handles [NotificationRemovedFromChannelEvent]. It makes a request to API to define outcome of handling.
+     *
+     * @param event Instance of [NotificationRemovedFromChannelEvent] that is being handled.
+     * @param filter [FilterObject] which is used to define an outcome.
+     */
+    override fun onNotificationRemovedFromChannelEvent(
+        event: NotificationRemovedFromChannelEvent,
+        filter: FilterObject,
+    ): EventHandlingResult = handleCidEventByRequest(event.cid, filter)
 
-    private fun handleCidEventByRequest(event: HasChannel, filter: FilterObject): EventHandlingResult {
+    private fun handleCidEventByRequest(cid: String, filter: FilterObject): EventHandlingResult {
         return runBlocking {
-            if (newChannelEventFilter(event.channel, filter)) {
+            if (newChannelEventFilter(cid, filter)) {
```

Comment:
I think we should rename this filter to something more understandable, like `isNewChannelAdded` or something like that.

Also, maybe we can have the channel IDs that we have active stored in a `Set` so that when we do a `contains` check it's much faster than checking if `any` channel ID matches the new ID.

Reply:
I didn't get an idea with sorting

Reply:
I was proposing storing a Set of all channel IDs that are active and then using those to quickly check if the new ID is within that set.

But this is fine too!